drivers/volume/portworx/portworx.go

```diff
@@ -642,22 +643,38 @@ func (d *portworx) GetNodeForVolume(vol *torpedovolume.Volume) (*node.Node, erro
 		}
 	}
 
-	pxVol := v.(*api.Volume)
-	for _, n := range node.GetStorageDriverNodes() {
-		if n.VolDriverNodeID == pxVol.AttachedOn {
-			return &n, nil
+	r := func() (interface{}, bool, error) {
+		pxVol := v.(*api.Volume)
+		for _, n := range node.GetStorageDriverNodes() {
+			if n.VolDriverNodeID == pxVol.AttachedOn {
+				return &n, false, nil
+			}
+		}
+
+		// Snapshots may not be attached to a node
+		if pxVol.Source.Parent != "" {
+			return nil, false, nil
 		}
+
+		err := fmt.Errorf("Volume: %s is not attached on any node", name)
+		logrus.Warnf(err.Error())
```

Comment:
The `logrus.Warnf(err.Error())` is not needed. The task.DoRetryWithTimeout should print the error.

Reply:
Removed logrus.Warnf() print and just returning fmt.Errorf()